src/librustdoc/json/mod.rs

```diff
@@ -69,7 +69,21 @@ impl JsonRenderer<'tcx> {
                     .iter()
                     .filter_map(|i| {
                         let item = &i.impl_item;
-                        if item.def_id.is_local() {
+
+                        // HACK(hkmatsumoto): For impls of primitive types, we index them
+                        // regardless of whether they're local. This is because users can
+                        // document primitive items in an arbitrary crate by using
```

Comment:
> in an arbitrary crate by using

Is "_from_ an arbitrary crate by using" better here? I'm not a native speaker ;)

Reply:
I think "in" is better - primitives are available no matter what, it's only the documentation that can be in a crate.